server/src/main/java/org/elasticsearch/action/get/TransportGetAction.java

```diff
@@ -192,12 +192,7 @@ private void handleGetOnUnpromotableShard(GetRequest request, IndexShard indexSh
         throws IOException {
         ShardId shardId = indexShard.shardId();
         if (request.refresh()) {
-            var node = getCurrentNodeOfPrimary(clusterService.state(), shardId);
-            if (node == null) {
-                listener.onFailure(new NoShardAvailableActionException(shardId, "primary shard is not active"));
-                return;
-            }
```

Comment:
I think this (and the similar check in mget) is not needed here since the refresh call should properly fail the listener.